podcasts/TranscriptsDataRetriever.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+import PocketCastsUtils
+import PocketCastsServer
+import PocketCastsDataModel
+
+actor TranscriptsDataRetriever {
+
+    typealias Transcript = Episode.Metadata.Transcript
+    typealias TranscriptFormat = Episode.Metadata.TranscriptFormat
+
+    private var dataRequestMap: [URL: Task<Data, Error>] = [:]
+
+    private let cache: URLCache
+
+    public init() {
+        cache = URLCache(memoryCapacity: 1.megabytes, diskCapacity: 100.megabytes, diskPath: "transcripts")
+    }
+
+    public func loadBestTranscript(from transcripts: [Transcript]) async throws -> String? {
+        guard let transcript = Episode.Metadata.TranscriptFormat.bestTranscript(from: transcripts),
+              let url = URL(string: transcript.url) else {
+            return nil
+        }
+
+        return try await loadTranscript(url: url)
+    }
+
+    public func loadTranscript(url: URL) async throws -> String? {
+        let request = URLRequest(url: url)
+
+        if let cachedResponse = cache.cachedResponse(for: request),
+           let result = String(data: cachedResponse.data, encoding: .utf8),
+           !result.trim().isEmpty {
+            FileLog.shared.addMessage("Transcripts Data Retriever: returning cached data for transcript")
+            Task {
+                // trigger a background refresh to force cache update
+                try? await loadTranscriptFromServer(url, previousResponse: cachedResponse)
+            }
+            return result
+        }
+
+        return try await loadTranscriptFromServer(url)
+    }
+
+    private func loadTranscriptFromServer(_ url: URL, previousResponse: CachedURLResponse? = nil) async throws -> String? {
+        if let task = dataRequestMap[url] {
+            return try await String(data: task.value, encoding: .utf8)
+        }
+        FileLog.shared.addMessage("Transcripts Data Retriever: requesting transcript data \(url)")
+
+        let task = Task<Data, Error> {
+            do {
+                var request = URLRequest(url: url, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData)
```

Comment:
By using `reloadIgnoringLocalAndRemoteCacheData` this request sends a `cache-control: no-cache`. In my testing this doesn't make any difference but I think we should not send it, so maybe `reloadRevalidatingCacheData` might be a better choice here.

Reply:
Changed here: https://github.com/Automattic/pocket-casts-ios/pull/1955/commits/a919037a2e0d26277b1ff416f0a3c48e47ffc2f2